<?xml version="1.0"?>

<launch>

  <!-- OpenSim IK Solver -->
  <include file="$(find hiros_opensim_ik_solver)/launch/hiros_opensim_ik_solver_default.launch">
    <arg name="node_required" value="false"    />
    <arg name="node_name"     value="opensim" />

    <!-- General Parameters -->
    <arg name="n_threads"                    value="6"           />
    <arg name="input_topic"                  value="/xsens/osg"  />
    <arg name="sensor_to_opensim_rotation_x" value="-1.5707963"  />
    <arg name="sensor_to_opensim_rotation_y" value="1.5707963"   />
    <arg name="sensor_to_opensim_rotation_z" value="0"           />

    <!-- IMU Placer Parameters -->
    <arg name="model_path"                 value="$(find hiros_opensim_ik_solver)/models/human18dof_subj0_opensim.osim" />
    <arg name="perform_model_calibration"  value="true"          />
    <arg name="perform_heading_correction" value="false"         />
    <arg name="base_imu_label"             value="imu_pelvis_fl" />
    <arg name="base_heading_axis"          value="x"             />
    <arg name="save_calibrated_model"      value="false"         />
    <arg name="visualize_calibration"      value="false"         />

    <!-- IK Tool Parameters -->
    <arg name="accuracy"         value="1e-4" />
    <arg name="visualize_motion" value="true" />
  </include>


  <!-- Converter -->
  <node pkg="xsens_mimu_array_to_orientation_skeleton_group" type="xma2osg" name="converter" output="screen" required="true">
    <param name="node_name"    type="str" value="converter"   />

    <!-- Parameters -->
    <param name="input_topic"  type="str" value="/xsens/data" />
    <param name="output_topic" type="str" value="/xsens/osg"  />
    <param name="global_frame" type="str" value="world"       />
  </node>


  <!-- Rosbag -->
  <node pkg="rosbag" type="play" name="player" output="screen" required="true" args="--clock --quiet --rate 1 $(find hiros_opensim_ik_solver)/bags/2020-06-26-13-05-03_filtered.bag" />


  <!-- Visualize -->
  <node pkg="rqt_image_view" type="rqt_image_view" name="viewer" output="screen" required="false" args="/kinect/qhd/image_color_rect/compressed" />
</launch>





